raiden/transfer/node.py

```diff
@@ -171,6 +172,9 @@ def subdispatch_to_paymenttask(node_state, state_change, secrethash):
                 )
                 events = sub_iteration.events
 
+        if sub_iteration and sub_iteration.new_state is None:
+            del node_state.payment_mapping.secrethashes_to_task[secrethash]
```

Comment:
I'm having problems to understand why this is needed. Does `sub_iteration.new_state == None` signal that the payment is finished and we subsequently clean up the task from the `secrethashes_to_task` mapping? Is there more stuff that should be removed in this case?

Reply:
> Does sub_iteration.new_state == None signal that the payment is finished and we subsequently clean up the task from the secrethashes_to_task mapping?

Yes, the task finished, either successfully or not, so we need to clean up after it.]

> Is there more stuff that should be removed in this case?

Not that i recall.

Reply:
@LefterisJP could you also do the cleanup for the functions

- subdispatch_initiatortask
- subdispatch_mediatortask
- subdispatch_targettask

Reply:
> Yes, the task finished, either successfully or not, so we need to clean up after it.]

Are possible errors handled somewhere else?

Reply:
@palango Paul this is to avoid this assertion: https://gist.github.com/LefterisJP/9349fdc10be00c0f195ecb2222c561f7